os/mm/mm_heap/mm_getheap.c

```diff
@@ -51,9 +51,94 @@ extern struct mm_heap_s g_mmheap[CONFIG_MM_NHEAPS];
  * Pre-processor Definitions
  ****************************************************************************/
 
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+typedef struct app_heap_s {
+	struct mm_heap_s *heap;
+	struct app_heap_s *next;
+	struct app_heap_s *prev;
+} app_heap_s;
+
+static app_heap_s *app_heap_list;
+extern struct mm_heap_s g_pheap;
+#endif
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+void mm_add_app_heap_list(struct mm_heap_s *heap)
+{
+	app_heap_s *node = (app_heap_s *)kmm_malloc(sizeof(app_heap_s));
+	if (!node)
+	{
+		mdbg("Error allocating heap node\n");
+		return;
+	}
+
+	node->heap = heap;
+
+	/* Add the new heap node to the head of the list*/
+	node->next = app_heap_list;
```

Comment:
I cannot use dq_xxx api because, I need functionality of a list here. 
For example if i want to search an element using queue, it will look like below which is bit complicated. 

x = dq_rem_first()
compare x->element == value?
else
dq_add_last(x)
repeat till end of queue
But how to find end of queue??

Its a bit confusing and not neat for our usecase.

Reply:
I think we can with dq_peek and dq_rem.
for example,
```
node = dq_peek(QUEUE);
while (node != NULL) {
  
  if (node == XXX) {
      XXXXXX;
      return/break;
  }
  node = dq_next(node);
}
```

Reply:
Yes. You are right. 
I changed it.